src/special/build_runner.zig

```diff
@@ -0,0 +1,89 @@
+const root = @import("build.zig");
+const std = @import("std");
+const io = std.io;
+const fmt = std.fmt;
+const Builder = std.build.Builder;
+const Pkg = std.build.Pkg;
+const LibExeObjStep = std.build.LibExeObjStep;
+const ArrayList = std.ArrayList;
+///! This is a modified build runner to extract information out of build.zig
+///! Modified from std.special.build_runner
+
+// We use a custom Allocator to intercept the creation of steps
+const InterceptAllocator = struct {
```

Comment:
Getting a list of the `LibExeObjStep`s is a little more involved than I would like but there is no way to differentiate between steps.  
The only way would be to check the make functions afainst the step function we want but they are not `pub`.

Reply:
I don't think differentiating between steps *necessarily* matters as we're just looking to resolve library names to paths, and this works flawlessly for doing so!

Reply:
We can definitely change std/build.zig code to make this use case easier! No need for such a hack when upstream is willing to improve the API :)